src/Quartz/CronExpression.cs

```diff
@@ -433,6 +433,7 @@ protected CronExpression(SerializationInfo info, StreamingContext context)
                     ThrowHelper.ThrowNotSupportedException($"Unknown serialization version {version}");
                     break;
             }
+            BuildExpression(CronExpressionString);
```

Comment:
Because there's already `OnDeserialization` implemented, doesn't it make the logic run twice wasting cycles in non-JSON context?

Reply:
1) Yes, I can confirm what you say that the existing serialization/deserilization will trigger OnDeserilization method, so running BuildExpression twice, (bad).

2) From what I read, System.Text.json and it appears newtonsoft default settings, do not trigger OnDeserialization.  Therefore should a person utilise these to serialize/deserialize the OnDeserilization method are not called. (https://stackoverflow.com/questions/67222737/ondeserialized-is-not-called-in-net-core)

3) can I propose two options:

a) Drop `OnDeserializaiton`, given that it only calls `BuildExpression`, and keep the PR'd BuildExpression in `CronExpression(SerializationInfo info, StreamingContext context)`.

b) Keep `OnDeserialization` as is, utilise the existing `bool expressionParsed` property (which appears unused) to flag Build has occured and conditionally skip in `protected CronExpression(SerializationInfo info, StreamingContext context)`

Thoughts?

Reply:
maybe go with A, removing the callback/interface would be breaking change but as this is gargeting main I guess it's fine

Reply:
ok done.